frontend/public/components/cluster-settings/global-config.tsx

```diff
@@ -49,25 +35,49 @@ const ItemRow = ({ item }) => {
   );
 };
 
-class GlobalConfigPage_ extends React.Component<GlobalConfigPageProps, GlobalConfigPageState> {
-  readonly state: GlobalConfigPageState = {
-    errors: [],
-    items: [],
-    loading: true,
-    textFilter: '',
-  };
-
-  componentDidMount() {
+const GlobalConfigPage_: React.FC<GlobalConfigPageProps & GlobalConfigPageExtensionProps> = ({
+  configResources,
+  globalConfigs,
+}) => {
+  const [errors, setErrors] = React.useState([]);
+  const [items, setItems] = React.useState([]);
+  const [loading, setLoading] = React.useState(true);
+  const [textFilter, setTextFilter] = React.useState('');
+  const { t } = useTranslation();
+
+  const oauthMenuItems = _.map(addIDPItems, (label: string, id: string) => ({
+    // t('oauth~Basic Authentication')
+    // t('oauth~GitHub')
+    // t('oauth~GitLab')
+    // t('oauth~Google')
+    // t('oauth~HTPasswd')
+    // t('oauth~Keystone')
+    // t('oauth~LDAP')
+    // t('oauth~OpenID Connect')
+    // t('oauth~Request Header')
```

Comment:
https://github.com/openshift/console/blob/master/frontend/public/components/cluster-settings/oauth.tsx#L54-L64 is just an object.  Not sure how I'd do that.

Reply:
> https://github.com/openshift/console/blob/master/frontend/public/components/cluster-settings/oauth.tsx#L54-L64 is just an object. Not sure how I'd do that.

I believe it can be anywhere in any file, and the parser will find it.